doc/source/ray-air/preprocessors.rst

```diff
@@ -0,0 +1,175 @@
+.. _air-preprocessors:
+
+Preprocessing Data
+==================
+
+This page describes how to perform data preprocessing in Ray AIR.
+
+Data preprocessing is a common technique for transforming raw data into features that will be input to a machine learning model.
+In general, you may want to apply the same preprocessing logic to your offline training data and online inference data.
+Ray AIR provides several common preprocessors out of the box as well as interfaces that enable you to define your own custom logic.
+
+Overview
+--------
+
+Ray AIR exposes a ``Preprocessor`` class for preprocessing. The Preprocessor has 4 methods that make up its core interface.
+
+#. ``fit()``: Compute some state for a Dataset and save it to the Preprocessor.
+#. ``transform()``: Apply a transformation to a Dataset.
+#. ``transform_batch()``: Apply a transformation to a single batch of data.
+#. ``fit_transform()``: Syntactic sugar for calling both ``fit()`` and ``transform()`` on a Dataset.
+
+To show these in action, let's walk through a basic example. First we'll set up 2 simple Ray Datasets.
+
+.. literalinclude:: doc_code/preprocessors.py
+    :language: python
+    :start-after: __preprocessor_setup_start__
+    :end-before: __preprocessor_setup_end__
+
+Next, ``fit`` the Preprocessor on one Dataset, and ``transform`` both Datasets with this fitted information.
+
+.. literalinclude:: doc_code/preprocessors.py
+    :language: python
+    :start-after: __preprocessor_fit_transform_start__
+    :end-before: __preprocessor_fit_transform_end__
+
+Finally, call ``transform_batch`` on a single batch of data.
+
+.. literalinclude:: doc_code/preprocessors.py
+    :language: python
+    :start-after: __preprocessor_transform_batch_start__
+    :end-before: __preprocessor_transform_batch_end__
+
+Life of an AIR Preprocessor
+---------------------------
+
+Now that we've gone over the basics, let's dive into how Preprocessors fit into an end-to-end application built with AIR.
+
+.. figure:: images/air-preprocessor.svg
+
+Throughout this section, we'll go through an example using XGBoost. The same logic is applicable to other integrations as well.
+
+Trainer
+~~~~~~~
+
+The journey of the Preprocessor starts with the ``Trainer``. If the Trainer is instantiated with a Preprocessor,
+then the following logic will be executed when ``Trainer.fit()`` is called:
+
+#. If a ``"train"`` Dataset is passed in, then the Preprocessor will call ``fit()`` on it.
+#. The Preprocessor will then call ``transform()`` on *all* Datasets, including the ``"train"`` Dataset.
+#. The Trainer will then perform training on the preprocessed Datasets.
+
+.. literalinclude:: doc_code/preprocessors.py
+    :language: python
+    :start-after: __trainer_start__
+    :end-before: __trainer_end__
+
+.. note::
```

Comment:
hmm, should we introduce a `.. todo::` directive so that we don't forget to update this once the issue is resolved? I'm always worried about such comments becoming stale.

Reply:
Added a `.. TODO:` (`.. todo::` raises an error!)